### PR TITLE
(dbt) Firebase pro migration

### PR DESCRIPTION
**Pull request by mathilde-vey**
# DA PR

## Describe your changes

The purpose of this PR is to migrate the `firebase_pro_events` model as well as to create the SQL that will be used for exporting data into Posthog (for the firebase pro part).

## Jira ticket number and/or notion link

JIRA-ticket_number

### Type of change
- [ ] Fix (non-breaking change which corrects expected behavior)
- [ ] New fields (non-breaking change)
- [x] New table (non-breaking change)
- [ ] Concept change (potentially breaking change which modifies fields according to new or evolving business concepts) 
- [ ] Table deletion (potentially breaking change which adds functionality/ table)
      
### Checklist before requesting a review
- [x] I have performed a self-review of my code
- [x] Fields have been snake_cased
- [ ] I have checked my modifications don't break downstream models
- [ ] If my changes concern incremental table, I have altered their schema to accomodate with field's creation/deletion
- [ ] I have made corresponding changes to the [tables documentation](https://www.notion.so/passcultureapp/Documentation-Tables-175a397a8e854ff4a55ae4f3620dbe3b)
- [ ] I have made corresponding changes to the [fields glossary](https://www.notion.so/passcultureapp/854a436a8f1541e1b6ec2a65f8bab600?v=798024ba90404b139e5a17407a3bc604)
- [ ] I have updated the dag in cases of dependencies
- [ ] My code passes CI/CD tests
- [ ] I will post on slack review channel and ensure to specify the duration of the review task: short (<10min), medium (<30min), long (>30min)

### CODE REVIEW

File: orchestration/dags/data_gcp_dbt/models/intermediate/firebase/native/int_firebase__native_event.sql [hunk 1]
```diff
@@ -91,7 +91,7 @@ SELECT
     CASE WHEN firebase_screen = "SignupConfirmationEmailSent" OR event_name = "ContinueCGU" THEN 1 ELSE 0 END AS is_signup_completed,
     CASE WHEN firebase_screen IN ("BeneficiaryRequestSent","UnderageAccountCreated","BeneficiaryAccountCreated") THEN 1 ELSE 0 END AS is_benef_request_sent,
     CASE WHEN event_name = "login" THEN 1 ELSE 0 END AS is_login
-FROM {{ ref("int_firebase__event_flattened") }}
+FROM {{ ref("int_firebase__native_event_flattened") }}
```
Comment (carolinemestre-passculture) [inline orchestration/dags/data_gcp_dbt/models/intermediate/firebase/native/int_firebase__native_event.sql:94]: @MatthieuRipollPass, par rapport aux namings des modèles firebase. J'ai update le dossier (créer un dossier native / pro dans le dossier firebase) + le nom des modèles. Tu me diras ce que tu en penses.

Comment (cdelabre) [reply orchestration/dags/data_gcp_dbt/models/intermediate/firebase/native/int_firebase__native_event.sql:94]: Top.

File: orchestration/dags/data_gcp_dbt/models/mart/post_hog/mrt_post_hog__firebase_pro_event.sql [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+        partner_id,
+        partner_name,
+        partner_type,
+        partner_cultural_sector,
+        partner_nb_individual_offers,
+        partner_nb_collective_offers
+    ) as user_params,
+    "pro" as origin
+FROM {{ ref("mrt_global__firebase_pro_event") }}
+WHERE MOD(ABS(FARM_FINGERPRINT(user_pseudo_id)),10) = 0
```
Comment (cdelabre) [inline orchestration/dags/data_gcp_dbt/models/mart/post_hog/mrt_post_hog__firebase_pro_event.sql:51]: No need

File: orchestration/dags/data_gcp_dbt/models/mart/post_hog/mrt_post_hog__firebase_pro_event.sql
Comment (cdelabre) [inline orchestration/dags/data_gcp_dbt/models/mart/post_hog/mrt_post_hog__firebase_pro_event.sql:1]: Cette table n'a pas besoin d'exister dans `l'environnement mart` mais on peut garder comme ça pour le moment.

**Review by cdelabre — DISMISSED** [review]

File: orchestration/dags/data_gcp_dbt/models/mart/post_hog/mrt_post_hog__firebase_pro_event.sql (hunk 2 above)
Comment (carolinemestre-passculture) [reply orchestration/dags/data_gcp_dbt/models/mart/post_hog/mrt_post_hog__firebase_pro_event.sql:51]: @mathilde-vey go on le retire? 
```suggestion
```

File: orchestration/dags/data_gcp_dbt/models/mart/post_hog/mrt_post_hog__firebase_pro_event.sql
Comment (cdelabre) [reply orchestration/dags/data_gcp_dbt/models/mart/post_hog/mrt_post_hog__firebase_pro_event.sql:1]: Pas vraiment la même table ici, car on met les données en STRUCT de manière à garder une convention d'export pour posthog avec adage et native.
Mais sinon oui pour mrt_global__firebase_pro_event.sql

File: orchestration/dags/data_gcp_dbt/models/intermediate/firebase/pro/int_firebase__pro_session.sql [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+SELECT
+  DISTINCT e.unique_session_id,
+  e.user_id
+  ro.offerer_id,
+  rv.venue_id
+FROM {{ ref("int_firebase__pro_event") }} e
+LEFT JOIN ranked_offers AS ro ON e.unique_session_id = ro.unique_session_id
+LEFT JOIN ranked_venues AS rv ON e.unique_session_id = rv.unique_session_id
+WHERE (ro.rn = 1 OR ro.rn IS NULL)
+AND (rv.rn = 1 OR rv.rn IS NULL)
```
Comment (cdelabre) [inline orchestration/dags/data_gcp_dbt/models/intermediate/firebase/pro/int_firebase__pro_session.sql:32]: Use QUALIFY for clarity in `WITH AS clause`

File: orchestration/dags/data_gcp_dbt/models/mart/post_hog/mrt_post_hog__firebase_pro_event.sql
Comment (carolinemestre-passculture) [reply orchestration/dags/data_gcp_dbt/models/mart/post_hog/mrt_post_hog__firebase_pro_event.sql:1]: Oui ok j'ai confondu entre les 2 ! Pour Post Hog on ne fait que des views donc on stock juste les queries8

File: orchestration/dags/data_gcp_dbt/models/intermediate/firebase/pro/int_firebase__pro_event.sql [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+        REGEXP_EXTRACT(page_location, r"""passculture\.pro\/.*?\/.*?\/(.*?)\/|\?""", 1) as url_path_details,
+        REGEXP_EXTRACT(page_location, r"""passculture\.pro\/(.*?)\?""", 1) as url_path_before_params,
+        REGEXP_EXTRACT_ALL(page_location,r'(?:\?|&)(?:([^=]+)=(?:[^&]*))') as url_params_key,
+        REGEXP_EXTRACT_ALL(page_location,r'(?:\?|&)(?:(?:[^=]+)=([^&]*))') as url_params_value,
+        REGEXP_REPLACE(REGEXP_REPLACE(page_location , "[A-Z \\d]+[\\?\\/\\&]?", ""), "https://passculture.pro/", "") as url_path_agg,
+        page_referrer,
+        page_number,
+        COALESCE(double_offer_id,offerId) AS offer_id,
+        offerType as offer_type,
+        used,
```
Comment (cdelabre) [inline orchestration/dags/data_gcp_dbt/models/intermediate/firebase/pro/int_firebase__pro_event.sql:54]: We could rename theses columns to be a bit clearer on what the might be (used/saved/ eac_wrong_student_modal_only6and5)

File: orchestration/dags/data_gcp_dbt/models/mart/global/mrt_global__firebase_pro_event.sql [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+        on_schema_change = "sync_all_columns"
+    )
+}}
+
+WITH most_active_offerer_per_user AS(
+SELECT 
+    DISTINCT uo.user_id
+    ,uo.offerer_id
+FROM {{ ref("enriched_user_offerer") }} AS uo 
+LEFT JOIN {{ ref("enriched_offerer_id") }} AS o ON uo.offerer_id=o.offerer_id
```
Comment (carolinemestre-passculture) [inline orchestration/dags/data_gcp_dbt/models/mart/global/mrt_global__firebase_pro_event.sql:15]: @mathilde-vey c'est une erreur de référence ici non? Tu voulais appeler quel modèle?

**Pull request merged**